testing/test_config.py

```diff
@@ -365,6 +365,52 @@ def pytest_addoption(parser):
         config = testdir.parseconfig()
         assert config.getini("strip") is bool_val
 
+    def test_addini_choices(self, testdir):
+        testdir.makeconftest(
+            """
+            def pytest_addoption(parser):
+                parser.addini("mychoice", "", type="choice", default="def", choices=["one", "two"])
+        """
+        )
+        config = testdir.parseconfig()
+        assert config.getini("mychoice") == "def"
+
+        testdir.makeini(
+            """
+            [pytest]
+            mychoice=invalid
+        """
+        )
+        config = testdir.parseconfig()
+        with pytest.raises(UsageError, match="invalid value for mychoice: 'invalid'"):
+            config.getini("mychoice")
+
+        p1 = testdir.makepyfile(
+            """
+            def test_pass(pytestconfig):
+                ini_val = pytestconfig.getini("mychoice")
+                print('\\nmychoice:%s\\n' % ini_val)"""
+        )
+        result = testdir.runpytest("-s", str(p1))
+        result.stdout.fnmatch_lines(
+            ["E *UsageError: invalid value for mychoice: 'invalid'", "*= 1 failed in *"]
+        )
+        assert result.ret == ExitCode.TESTS_FAILED
+
+        p1 = testdir.makepyfile(
```

Comment:
I usually prefer to set the data in some module (say `sys`) and inspect it later:

``` 
        monkeypatch.setattr(sys, "MY_INI_VAL", None, raising=False)
        p1 = testdir.makepyfile(
            """
            import sys
            def test_pass(pytestconfig):
                ini_val = pytestconfig.getini("mychoice")
                sys.MY_INI_VAL = pytestconfig.getini("mychoice")
                print('\\nmychoice:%s\\n' % ini_val)"""
        )
        result = testdir.runpytest("-s", "-o", "mychoice=one", str(p1))
        assert sys.MY_INI_VAL == "one"
        assert result.ret == ExitCode.OK
```

But it is merely a comment in case you like this approach.